web-components/src/components/table-advanced/TableAdvanced.ts

```diff
@@ -68,20 +71,47 @@ export namespace TableAdvanced {
     connectedCallback() {
       super.connectedCallback();
       document.addEventListener("dragover", this.dragover);
+      this.updateTableConfig();
+      this.updateTableData();
       this.populateTable();
     }
 
     protected update(changedProperties: PropertyValues) {
       super.update(changedProperties);
-      if (changedProperties.has("data")) {  
-         this.updateDataInTable();
+      if (changedProperties.has("data")) {
+        this.updateTableData();
+        this.updateDataInTable();
+      }
+    }
+
+    private updateTableData() {
+      if (typeof this.data === "string") {
+        try {
+          this.tableData = JSON.parse(this.data);
+        } catch (error) {
+          console.error("[INVALID_JSON_FORMAT] Please recheck the config json");
+        }
+      } else {
+        this.tableData = { ...this.data };
+      }
+    }
+
+    private updateTableConfig() {
+      if (typeof this.config === "string") {
+        try {
+          this.tableConfig = JSON.parse(this.config);
+        } catch (error) {
+          console.error("[INVALID_JSON_FORMAT] Please recheck the data json");
+        }
+      } else {
+        this.tableConfig = { ...this.config };
       }
-    } 
+    }
 
     private populateColumns() {
       let index = 0;
       const pushCol = (col: ColOptions, group?: { name: string; length: number }) => {
-        const f = col.filters || this.config.default?.col?.filters;
```

Comment:
is this `config` property still in use elsewhere? It seems like it is still initialized as a prop in the component (like, not removed or replaced explicitly by tableConfig)

Reply:
It has been replaced everywhere in the logic part of the code so what I am doing is user will still send config and data to props named config and data so as to ensure nothing breaks wherever it is currently running. After that I am taking the value from config and checking if it is string or not and assigning it to tableConfig and similarly to tableData. Then I am using tableConfig and tableData everywhere

Reply:
Ah, so kinda to prevent a breaking change it sounds like. Cool